gradle.groovy

```diff
@@ -0,0 +1,102 @@
+import groovy.json.JsonSlurper
+
+def doesUnimoduleSupportPlatform(Map unimoduleJson, String platform) {
+  def platforms = unimoduleJson.platforms// || ['android']
+
+  if (platforms instanceof Map) {
+    return platforms.containsKey(platform)
+  }
+  if (platforms instanceof List) {
+    return platforms.contains(platform)
+  }
+  return true
+}
+
+def doesUnimoduleSupportTarget(Map unimoduleJson, String target) {
+  def targets = unimoduleJson.targets
+  return !targets || targets.contains(target)
+}
+
+def findUnimodules(String target, List exclude, List modulesPaths) {
+  def unimodules = []
+
+  for (modulesPath in modulesPaths) {
+    def moduleConfigPaths = new FileNameFinder().getFileNames(modulesPath, '**/unimodule.json', '')
+
+    for (moduleConfigPath in moduleConfigPaths) {
+      def unimoduleConfig = new File(moduleConfigPath)
+      def unimoduleJson = new JsonSlurper().parseText(unimoduleConfig.text)
+      def directory = unimoduleConfig.getParent()
+      
+      if (doesUnimoduleSupportPlatform(unimoduleJson, 'android') && doesUnimoduleSupportTarget(unimoduleJson, target)) {
+        def unimoduleName = unimoduleJson.name
+
+        if (!exclude.contains(unimoduleName)) {
+          unimodules.add([
+            name: unimoduleJson.name,
+            directory: directory,
+            config: unimoduleJson.platforms instanceof Map && unimoduleJson.platforms,
+          ])
+        }
+      }
+    }
+  }
+
+  return unimodules
+}
+
+ext.useUnimodules = { Map customOptions = [] ->
+  def options = [
+    modulesPaths: ['../node_modules'],
+    configuration: 'expendency',
+    target: 'react-native',
+    exclude: [],
+  ] << customOptions
+
+  def unimodules = findUnimodules(options.target, options.exclude, options.modulesPaths)
+
+  for (unimodule in unimodules) {
+    if (options.configuration == 'expendency') {
+      expendency(unimodule.name)
+    } else {
+      Object dependency = project.project(":${unimodule.name}")
```

Comment:
This loosely typed `Object dependency` looks strange, maybe it would look better as the method argument? 🤔